app/internal/security/ouath2.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime, timedelta
+from typing import Union
+
+from app.config import JWT_ALGORITHM, JWT_KEY, JWT_MIN_EXP
+from passlib.context import CryptContext
+from app.database.models import User
+from fastapi import Depends, HTTPException
+from fastapi.security import OAuth2PasswordBearer
+import jwt
+from jwt.exceptions import InvalidSignatureError
+from sqlalchemy.orm import Session
+from starlette.requests import Request
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_401_UNAUTHORIZED
+from .schema import LoginUser, CurrentUser
+
+
+pwd_context = CryptContext(schemes=["bcrypt"])
+oauth_schema = OAuth2PasswordBearer(tokenUrl="/login")
+
+
+async def get_db_user_by_username(db: Session, username: str) -> User:
+    """Checking database for user by username field"""
+    user = db.query(User).filter_by(username=username).first()
+    return user
+
+
+def get_hashed_password(password: str) -> str:
+    """Hashing user password"""
+    return pwd_context.hash(password)
+
+
+def verify_password(plain_password: str, hashed_password: str) -> bool:
+    """Verifying password and hashed password are equal"""
+    return pwd_context.verify(plain_password, hashed_password)
+
+
+async def authenticate_user(
+        db: Session, user: LoginUser) -> Union[LoginUser, bool]:
+    """Verifying user is in database and password is correct"""
+    db_user = await get_db_user_by_username(db=db, username=user.username)
+    if db_user:
+        if verify_password(user.password, db_user.password):
+            return LoginUser(
+                user_id=db_user.id,
+                username=user.username, password=db_user.password)
+    return False
+
+
+def create_jwt_token(
+        user: LoginUser, JWT_MIN_EXP: int = JWT_MIN_EXP,
+        JWT_KEY: str = JWT_KEY) -> str:
+    """Creating jwt-token out of user unique data"""
+    expiration = datetime.utcnow() + timedelta(minutes=JWT_MIN_EXP)
+    jwt_payload = {
+        "sub": user.username,
+        "user_id": user.user_id,
+        "exp": expiration}
+    jwt_token = jwt.encode(
+        jwt_payload, JWT_KEY, algorithm=JWT_ALGORITHM)
+    return jwt_token
+
+
+async def check_jwt_token(
+    db: Session,
+    token: str = Depends(oauth_schema),
+        path: bool = None) -> CurrentUser:
+    """
+    Check whether JWT token is correct. Returns User object if yes.
+    Returns None or raises HTTPException,
+    depanding which depandency activated this function.
+    """
+    try:
+        jwt_payload = jwt.decode(
+            token, JWT_KEY, algorithms=JWT_ALGORITHM)
+        jwt_username = jwt_payload.get("sub")
+        jwt_user_id = jwt_payload.get("user_id")
+        db_user = await get_db_user_by_username(db, username=jwt_username)
+        if db_user and db_user.id == jwt_user_id:
+            return CurrentUser(**db_user.__dict__)
```

Comment:
What is the difference between `CurrentUser` and `LoginUser`?

Reply:
LoginUser is a pydantic object for internal actions management  of the security system.
CurrentUser is the Object that will be returned to the routes from the dependencies functions, after verifying JWT token.
I'm thinking maybe to re-change it to a Base Model object, as the code is keep on changing by other. And this object has to be maintained in order to keep it's functionality.

Reply:
What are the downsides of having only one of them? Currently having two different classes for two similar objects is a bit confusing from the developer side

Reply:
If by 'them' you mean LoginUser and CurrentUser, so they are not the same at all. LoginUser is created from the login form, and verifies that required information was given by the user. It is essential for this action. As CurrentUser supposed to return all the User information to the routes, after jwt verification.
Maybe it will be better to delete CurrentUser, and just return the User Base Object? It's a db object, no confusions there.

Reply:
I think the major source of confusion is in the object's name. Maybe we should find a better name which describes it better and which highlights the differences between them.

Reply:
if it ok by you, i will just return db_user. otherwise, every change in User base model will have to be updated in CurrentUser, in order for it to return the new column.

Reply:
Just be sure to name stuff properly. Otherwise it'll be really PITA to maintain :)